data_model/src/lib.rs

```diff
@@ -600,6 +685,18 @@ where
     }
 }
 
+impl TryFrom<Value> for BlockValue {
```

Comment:
Consider deriving.

Reply:
There's no `TryFrom`, and for `BlockValue` we can't use `TtryInto` as it has too many fields.

Reply:
> we can't use TtryInto as it has too many fields.

what do you mean?

Reply:
First I thought that it doesn't work for complex structures (with more then 2 fields), but then this: https://jeltef.github.io/derive_more/derive_more/try_into.html#structs

Reply:
I'm adding a ticket to derive it.